### ISSUE TITLE
Collection filtering on identifiers gives empty result

### ISSUE THREAD

**Issue by copitz** [TomFrost/Jexl#38]
Hi,

thank you for this awesome library :)

I just noticed that:

```javascript
const context = {
  items: [
    {id: 123, title: 'Hello world'},
    {id: 456, title: 'Good bye world'}
  ],
  my: {
    itemId: 123
  }
}

// The following will resolve with an empty array:
jexl.eval('items[.id == my.itemId]', context).then(console.log)

// While this will give the correct subset:
jexl.eval('items[.id == (my.itemId)]', context).then(console.log)
```

**Comment by TomFrost** [comment]
Oh my! That's an alarming bug indeed... thanks for reporting! I'll look into this.

**Comment by TomFrost** [comment]
Apologies for the SUPER late fix -- but this is now corrected in version 2.0.1!
